### DISCUSSION TITLE
Add outgoing messages to history before succesfully sent or not

### DISCUSSION THREAD

**Discussion by fulalas** [d99kris/nchat#199] [Ideas]
I see 2 reasons for that:

1- if the internet connection is down, the messages we send go nowhere and it feels we should send them again, until the connection is back and we notice we sent many similar messages;
2- it's much more satisfying to have UX feedback with no delay, but at the moment it takes a few noticeable milliseconds between the shortcut to send a message and the history showing it;

**Comment by d99kris** [comment]
Hi @fulalas - so I assume you are seeing a noticeable delay? Is this for Telegram or WhatsApp? On a high latency network connection (like mobile) or a "normal" connection?

It's unfortunately a non-trivial change (nchat architecture currently relies on what the underlying protocol reports as sent before displaying), but I suppose it's something that can be looked into. It might be a bit low priority though..

**Comment by fulalas** [comment]
In both Telegram and WhatsApp I see similar latency using high speed desktop connection.

**Reply by d99kris** [reply]
Thanks 👍
